### ISSUE TITLE
allow the plugin work without internet

### ISSUE THREAD

**Issue by vn971** [intellij-rust/intellij-rust#1045]
EDIT:
until importing project does not succeed (like when you can't fetch crates.io), the Settings>Languages&>Rust section is broken. It is impossible to set the stdlib location. After I fill in the stdlib location, I cannot press the "Apply" button. It just won't press and stays enabled. If I click "OK" and return to this setting back, I see the stdlib location empty.

Old description:
<strike>
```text
Cargo execution failed (exit code 101).
								            /home/vasya/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo metadata --verbose --format-version 1 --all-features
								            stdout : 
								            stderr :  Downloading libsqlite3-sys v0.4.0
								warning: spurious network error (2 tries remaining): [6] Couldn't resolve host name (Couldn't resolve host 'crates.io')
								warning: spurious network error (1 tries remaining): [6] Couldn't resolve host name (Couldn't resolve host 'crates.io')
								error: unable to get packages from source
```

Also, related or not, I absolutely cannot change the Standard library location. I go to Settings > Languages& > Rust. After I fill in the stdlib location, I cannot press the "Apply" button. It just won't press and stays enabled. If I click "OK" and return to this setting back, I see the stdlib location empty.
</strike>

**Comment by vn971** [comment]
Sorry, this was because I haven't really built the project on my computer. So it had no chance to fully work without internet connection.

There might be a bug worth fixing though in that idea could at least allow me setting stdlib, but I guess it's much less important then.

**Comment by vn971** [comment]
Feel free to close the issue if not deemed important.

**Comment by la10736** [comment]
Hi @vn971 , 
If you would like to work by cargo without internet connection you should create a local registry to replace `cargo.io`'s one. 

Take a look to https://github.com/alexcrichton/cargo-local-registry and https://github.com/rust-lang/cargo/pull/2361 to more dettails.

**Comment by vn971** [comment]
hi @la10736 . Thanks. Yes, I think I understood that part. The issue is about breaking the intellij UI while the cargo command still gives errors. Unclickable "Apply" button might be bad even for a broken configuration.
This is a minor issue though, because normally people don't (or shouldn't) break their cargo settings/repository.

**Comment by vn971** [comment]
I've updated the first comment in this issue to reflect the main current question.

**Comment by matklad** [comment]
@vn971 do you use `rustup`? If you, then you don't need to bother with configuring standard library, everything will be setup automatically.

The problem with "cargo does not work without Internet" is indeed real, but there's nothing obvious we can do to help the issue unfortunately. We can try to improve the error messages though...

**Comment by vn971** [comment]
@matklad I do have rustup. (I have it installed system-wide, idea can successfully pick it up on another project.)
Yeah, I guess some error messages and UI fixes might be the way to make it less strange. Like the "apply" button click-able, with some error or something (just not silently not working).

If it's not important, feel free to close the issue. IDK myself.

**Comment by matklad** [comment]
Superseded by https://github.com/intellij-rust/intellij-rust/issues/1069, thanks for the report!

**Comment by vn971** [comment]
@matklad sure, cool.
